src/scripts/generate-schemas.ts

```diff
@@ -18,19 +19,32 @@ interface SchemaProperty {
     owner: boolean;
 }
 
-interface Schema {
-    klass: string;
+interface Schema extends D2SchemaProperties {
     properties: SchemaProperty[];
-    name: string;
-    plural: string;
-    metadata: boolean;
     href: string;
 }
 
 interface Schemas {
     [className: string]: Schema;
 }
 
+const schemaProperties: Array<keyof D2SchemaProperties> = [
+    "klass",
+    "shareable",
+    "metadata",
+    "relativeApiEndpoint",
+    "plural",
+    "displayName",
+    "collectionName",
+    "nameableObject",
+    "translatable",
+    "identifiableObject",
+    "dataShareable",
+    "name",
+    "persisted",
+    "embeddedObject",
+];
+
```

Comment:
I'm not sure if we can do this using only the type, as types are not available on runtime....

Reply:
It's a typical scenario, the idiomatic is to defined the values and derive the type: 

```
const xs = ["hello", "bye"] as const
type Xs = (typeof xs)[number]
```